### PR TITLE
feat: implement sync multihash hasher

### PR DESCRIPTION
**Pull request by Gozala**
This is followup on https://github.com/multiformats/js-multiformats/issues/159 which would allow me to make sync HAMT implementation.

### CODE REVIEW

**Comment by Gozala** [comment]
@achingbrain looks like importer uses `encode` method that isn't part of `MultihashHasher` interface 

https://github.com/ipfs/js-ipfs-unixfs/blob/273a141b5ee3805bd0ef2dc8ed7870f8c6c8a820/packages/ipfs-unixfs-importer/src/options.js#L8-L16

In the second commit I've added encode functions back, yet I'm wondering if we need to just make `encode` part of the `MultihashHasher` API instead.

With this PR `import { murmur3128 } from @multiformats/murmur3` will not have a `encode` method, but `import * murmur3128 from '@multiformats/murmur3/murmur128'` will. I think this may be a right compromise.

File: murmur128.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export const name = 'murmur3-128'
+export const code = 0x22
+
+/**
+ * @param {Uint8Array} input
+ * @returns {import('multiformats/hashes/interface').MultihashDigest<typeof code>}
+ */
+export const digest = (input) =>
+  Digest.create(code, encode(input))
+
```
Comment (rvagg) [inline murmur128.js:14]:
```suggestion
export const digest = (input) => {
  return Digest.create(code, encode(input))
}

```

File: murmur128.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * @param {Uint8Array} input
+ */
+export const encode = (input) =>
+  bytes.fromHex(mur.x64.hash128(input))
```
Comment (rvagg) [inline murmur128.js:19]:
```suggestion
export const encode = (input) => {
  return bytes.fromHex(mur.x64.hash128(input))
}
```

File: murmur32.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export const name = 'murmur3-32'
+export const code = 0x23
+
+/**
+ * @param {Uint8Array} input
+ * @returns {import('multiformats/hashes/interface').MultihashDigest<typeof code>}
+ */
+export const digest = input =>
+  Digest.create(code, encode(input))
```
Comment (rvagg) [inline murmur32.js:12]:
```suggestion
export const digest = (input) => {
  return Digest.create(code, encode(input))
}
```

File: murmur32.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * @param {Uint8Array} input
+ */
+export const encode = input => fromNumberTo32BitBuf(mur.x86.hash32(input))
```
Comment (rvagg) [inline murmur32.js:17]:
```suggestion
export const encode = (input) => fromNumberTo32BitBuf(mur.x86.hash32(input))
```

**Review by rvagg — APPROVED** [review]
nice, just adding some formatting nits to satisfy my personal beefs about arrow function readability (something like: always use parens around arguments and multi-lines functions should not act like single-line ones [i.e. needs braces and returns])

**Comment by achingbrain** [comment]
> With this PR import { murmur3128 } from @multiformats/murmur3 will not have a encode method

This is a breaking change but this has gone out as a minor.  Can it please be reverted and released as v2?  It's [used by js-ipfs](https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core/src/components/files/utils/hamt-constants.js#L10) and will now start failing.

**Pull request merged**